### PR TITLE
Cleanup build matrix for matplotlib 3.2

### PR DESCRIPTION
**Pull request by hmaarrfk**
* Change the lower bound for numpy pinning
* Add a python 3.8 32 bit build.
* Remove repeated NP 1.14.5 pinning from most places
* Update numpy pinning for python 3.8

### CODE REVIEW

**Comment by hmaarrfk** [comment]
I'm mostly trying to get a 32 bit wheel created. Scikit-image publishes 32 bit wheels
https://github.com/scikit-image/scikit-image/issues/4314

**Comment by hmaarrfk** [comment]
Well, we are probably running into the fact that cryptography dropped support for 32 bit wheels.
https://github.com/pyca/cryptography/issues/5069

**Comment by hmaarrfk** [comment]
```
=================================== FAILURES ===================================

_____________________________ test_fancyarrow[png] _____________________________

[gw0] linux -- Python 3.7.0 /venv/bin/python3.7

expected = '/io/tmp_for_test/result_images/test_arrow_patches/fancyarrow_test_image-expected.png'

actual = PosixPath('/io/tmp_for_test/result_images/test_arrow_patches/fancyarrow_test_image.png')

tol = 0

    def _raise_on_image_difference(expected, actual, tol):

        __tracebackhide__ = True

    

        err = compare_images(expected, actual, tol, in_decorator=True)

        if err:

            for key in ["actual", "expected"]:

                err[key] = os.path.relpath(err[key])

            raise ImageComparisonFailure(

                'images not close (RMS %(rms).3f):\n\t%(actual)s\n\t%(expected)s '

>                % err)

E           matplotlib.testing.exceptions.ImageComparisonFailure: images not close (RMS 0.007):

E           	result_images/test_arrow_patches/fancyarrow_test_image.png

E           	result_images/test_arrow_patches/fancyarrow_test_image-expected.png

/venv/lib/python3.7/site-packages/matplotlib/testing/decorators.py:136: ImageComparisonFailure
```

Not sure, but at least now it compiles.

File: config.sh [hunk 1]
```diff
@@ -6,8 +6,12 @@ LOCAL_FT_COMMIT=5ad9b15
 
 # Test arguments
 NPROC=2
-PYTEST_ARGS="-ra --maxfail=1 --timeout=300 --durations=25 -n $NPROC"
+PYTEST_ARGS="-ra --maxfail=1000 --timeout=300 --durations=25 -n $NPROC"
 
+function pip_opts {
+    # Define extra pip arguments
```
Comment (hmaarrfk) [inline config.sh:12]: cryptography dropped 32 bit linux support. this encourages pip to use the latest wheel that satisfies the requirements.

**Comment by tacaswell** [comment]
@hmaarrfk Sorry, I did not notice that this PR was open and accidentally closed it when I deleted the target branch to standardize the names!

**Comment by hmaarrfk** [comment]
all good!

**Pull request closed without merging**